src/capsule/CapsuleSigner.ts

```diff
@@ -0,0 +1,219 @@
+import { USER_NOT_AUTHENTICATED_ERROR, USER_NOT_MATCHING_ERROR } from '@capsule/client'
+import { normalizeAddressWith0x } from '@celo/base/lib/address'
+import { CeloTx, RLPEncodedTx, Signer } from '@celo/connect'
+import { EIP712TypedData, generateTypedDataHash } from '@celo/utils/lib/sign-typed-data-utils'
+import { encodeTransaction, extractSignature, rlpEncodedTx } from '@celo/wallet-base'
+import { fromRpcSig } from 'ethereumjs-util'
+import { NativeModules } from 'react-native'
+import Logger from 'src/utils/Logger'
+import { PrivateKeyStorage } from './PrivateKeyStorage'
+import userManagementClient from './UserManagementClient'
+import { base64ToHex, hexToBase64 } from './helpers'
+
+const { CapsuleSignerModule } = NativeModules
+
+const TAG = 'geth/CapsuleSigner'
+
+/**
+ * Wrapper for request to refresh cookie and retry on cookies-related failures
+ * @param request request function
+ * @param reauthenticate function to refresh session cookies
+ */
+async function requestAndReauthenticate<T>(
+  request: () => Promise<T>,
+  reauthenticate: () => Promise<void>
+): Promise<T> {
+  try {
+    return await request()
+  } catch (e) {
+    const { data } = e.response
+    if (data === USER_NOT_MATCHING_ERROR || data === USER_NOT_AUTHENTICATED_ERROR) {
+      await reauthenticate()
+      return await request()
+    }
+    throw e
+  }
+}
+
+/**
+ * Implements the signer interface using the CapsuleSignerModule
+ */
+export abstract class CapsuleBaseSigner implements Signer {
+  private account: string | undefined
+  private readonly userId: string
+  private keyshareStorage: PrivateKeyStorage | undefined
+  private ensureSessionActive: () => Promise<void>
+
+  constructor(userId: string, ensureSessionActive: () => Promise<void>) {
+    this.userId = userId
+    this.ensureSessionActive = ensureSessionActive
+  }
```

Comment:
I haven't seen the `ReactNativeCapsuleWallet` initialized, is this a function that we shall determine?

Reply:
see contracts.ts